client/src/components/Header/components/nav-links.tsx

```diff
@@ -441,7 +441,7 @@ export class NavLinks extends Component<NavLinksProps, {}> {
                       })}
                       onClick={this.handleLanguageChange}
                       onKeyDown={this.handleLanguageMenuKeyDown}
-                      {...(index === locales.length - 1 && {
+                      {...(index === locales.length - 2 && {
```

Comment:
Hi @mmatsumoto1026, thanks for digging into this accessibility issue, and for your detailed discussion with @Sboonny and @bbsmooth in https://github.com/freeCodeCamp/freeCodeCamp/pull/46644.

Tested this locally and it does restore the functionality described in [this comment](https://github.com/freeCodeCamp/freeCodeCamp/pull/46644#issuecomment-1210022946). But I'm not sure if this is the best fix.

We may end up enabling other languages and hiding them, and it will be hard to remember to update this line, too.

What might be better is to do something like this so we're referencing the filtered `locales` array that was added in https://github.com/freeCodeCamp/freeCodeCamp/pull/47362:

```
{locales
  .filter(lang => !hiddenLangs.includes(lang))
  .map((lang, index, availableLocales) => (
    <li key={'lang-' + lang} role='none'>
      <button
        {...(clientLocale === lang && { 'aria-current': true })}
        className='nav-link nav-lang-menu-option'
        data-value={lang}
        {...(LangCodes[lang] && {
          lang: LangCodes[lang] as string
        })}
        onClick={this.handleLanguageChange}
        onKeyDown={this.handleLanguageMenuKeyDown}
        {...(index === availableLocales.length - 1 && {
          ref: this.lastLangOptionRef
        })}
        role='menuitem'
        tabIndex='-1'
      >
        {LangNames[lang]}
      </button>
    </li>
  ))}
```

Something like that should be easier to maintain. Or alternatively, `hiddenLangs` can be filtered out of `locales` towards the top of the file where its declared, and we can go back to using the `.map()` method like we did before https://github.com/freeCodeCamp/freeCodeCamp/pull/47362/files#diff-90c4bcf7900983772f4117b2260e7bc27081d9f597bb4307f1675774fad55451L430.

Reply:
Hello, [@scissorsneedfoodtoo](https://github.com/scissorsneedfoodtoo), thanks for your thoughtful comment.

I agree your suggestions.

So I've tested not only the code you showed above, but also using ` hiddenLangs` to filter out of `locales` towards the top of the file. 
And both are work correctly based on the comment which is for testing the keyboard accessibility feature.

Using ` hiddenLangs`  at the top of the file seems to be less code,  so I'd like to adopt it.
```suggestion
                      {...(index === locales.length - 1 && {
```
```javascript
const locales = availableLangs.client.filter(lang => !hiddenLangs.includes(lang));
```
```javascript
{locales
  .map((lang, index) => (
    <li key={'lang-' + lang} role='none'>
      <button
        {...(clientLocale === lang && { 'aria-current': true })}
        className='nav-link nav-lang-menu-option'
        data-value={lang}
        {...(LangCodes[lang] && {
          lang: LangCodes[lang] as string
        })}
        onClick={this.handleLanguageChange}
        onKeyDown={this.handleLanguageMenuKeyDown}
        {...(index === locales.length - 1 && {
          ref: this.lastLangOptionRef
        })}
        role='menuitem'
        tabIndex='-1'
      >
        {LangNames[lang]}
      </button>
    </li>
  ))}
```